addon/mixins/base.js

```diff
@@ -89,27 +88,35 @@ Semantic.BaseMixin = Ember.Mixin.create({
     };
   },
 
-  didInsertElement: function() {
+  initializeModule() {
     this.$()[this.get("module")](this.settings(this.get("module")));
+  },
+
+  didInsertElement() {
+    this._super(...arguments);
+    this.initializeModule();
 
     var _this = this;
     var properties = {};
+
     // Modules without setable properties
-    if (['accordion', 'nag'].indexOf(this.get('module')) === -1) {
-      properties = this.execute('set');
-    }
+    properties = this.execute('internal', 'set');
     var property;
 
-    for(property in properties) {
-      if (!properties.hasOwnProperty(property)) {
-        continue;
-      }
+    if (typeof properties === "object" && !Ember.isArray(properties)) {
+      for(property in properties) {
+        if (!properties.hasOwnProperty(property)) {
+          continue;
+        }
 
-      _this.addObserver(property, _this, _this.updateProperty(property));
+        if (this.hasOwnProperty(property)) {
+          _this.addObserver(property, _this, _this.updateProperty(property));
```

Comment:
I figured we only need to observe properties that are actually on the component. Previously we observed them all